annotation/src/main/java/io/smallrye/common/annotation/IdentifiedBy.java

```diff
@@ -0,0 +1,51 @@
+package io.smallrye.common.annotation;
+
+import javax.enterprise.util.AnnotationLiteral;
+import javax.inject.Qualifier;
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * String-based {@linkplain Qualifier qualifier}.
+ * Unlike {@link javax.inject.Named @Named}, this is a proper qualifier in CDI environment.
+ * <p>
+ * Identifier must always be provided.
+ * Unlike {@link javax.inject.Named @Named}, it is not deduced from the annotation use.
+ */
+@Qualifier
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.FIELD, ElementType.PARAMETER, ElementType.METHOD, ElementType.TYPE})
+public @interface IdentifiedBy {
```

Comment:
Feedback on the name would be greatly appreciated :-) See also the discussion in https://github.com/smallrye/smallrye-reactive-messaging/issues/1096

Reply:
Grammatically, if a simple-past-tense verb is what is desired in order to be consistent with `Named`, I'd recommend `IdentifiedAs` instead of `IdentifiedBy`; similarly (I think) to @mkouba, I don't think `IdentifiedBy` is perfectly clear that the value is itself the identifier.

Reply:
Naming is hard! And, as a non-native speaker whose fondness of English morphology and grammar has never been terribly high, I'll just take your word for it :-)

And, I don't like "identified by" (as well as "identified as") either. If we could have a single word, that would be so much better.

Reply:
Fair!  Other single-word, simple-past-tense verbs with similar meaning: `Called` (seems ambiguous with invocation-related terminology), `Labelled`, `Dubbed`, or `Entitled`.  No options are particularly appealing, and several are ambiguous; CDI took the good one 🙂.

Reply:
Yes! JPA took `@Id`, JUnit Jupiter took `@Tag`, and now we're here :-)

/me grumbles and renames `@IdentifiedBy` to `@Moniker` :rofl:

Reply:
How about `@Tagged`? Still different from JUnit `@Tag`. Or you don't want to use the same verb?

Reply:
My concern with `Tagged` is that is leads to a lookup with is a selector matching multiple tags (à la Kubernetes). 
It can be interesting, but at the same time much more complicated.

Reply:
I've thought about this a lot yesterday and I think I'll just rename it to `@Identifier`. It's not a verb in simple past tense, but that has never been a requirement nor a convention for qualifiers.

Reply:
Renamed to `@Identifier`.